src/cli/has_specs.rs

```diff
@@ -12,27 +12,30 @@ pub(crate) trait HasSpecs {
     fn packages(&self) -> Vec<&str>;
 
     fn specs(&self) -> miette::Result<IndexMap<PackageName, MatchSpec>> {
-        let mut map = IndexMap::with_capacity(self.packages().len());
-        for package in self.packages() {
-            let spec = MatchSpec::from_str(package, ParseStrictness::Lenient).into_diagnostic()?;
-            let name = spec.name.clone().ok_or_else(|| {
-                miette::miette!("could not find package name in MatchSpec {}", spec)
-            })?;
-            map.insert(name, spec);
-        }
-        Ok(map)
+        self.packages()
+            .iter()
+            .map(|package| {
+                let spec =
+                    MatchSpec::from_str(package, ParseStrictness::Strict).into_diagnostic()?;
```

Comment:
```suggestion
                    MatchSpec::from_str(package, ParseStrictness::Lenient).into_diagnostic()?;
```

Reply:
I will trust you on that :)

Reply:
I had just fixed that in #1852

Reply:
Ah. That would have been funny if i had immediately reverted that. Nice find 😁